homeassistant/helpers/intent.py

```diff
@@ -794,33 +797,23 @@ def __init__(
                 self.optional_slots[key] = value_schema
 
     @cached_property
-    def _slot_schema(self) -> vol.Schema:
-        """Create validation schema for slots (with extra required slots)."""
-        if self.slot_schema is None:
-            raise ValueError("Slot schema is not defined")
-
-        if self.required_slots or self.optional_slots:
-            slot_schema = {
-                **self.slot_schema,
-                **{
-                    vol.Required(key[0]): schema
-                    for key, schema in self.required_slots.items()
-                },
-                **{
-                    vol.Optional(key[0]): schema
-                    for key, schema in self.optional_slots.items()
-                },
-            }
-        else:
-            slot_schema = self.slot_schema
-
-        return vol.Schema(
-            {
-                key: SLOT_SCHEMA.extend({"value": validator})
-                for key, validator in slot_schema.items()
+    def effective_slot_schema(self) -> vol.Schema:
+        """Create validation schema for slots (with extra slots)."""
+        if not self.required_slots and not self.optional_slots:
+            return super().effective_slot_schema
+
+        extra_slots = {
+            **{
+                vol.Required(key[0]): schema
+                for key, schema in self.required_slots.items()
             },
-            extra=vol.ALLOW_EXTRA,
```

Comment:
It's critical that `ALLOW_EXTRA` is injected into the overall slot schema. There are slots that get automatically set based on context, such as the device's area. Not all intents support this, so they should ignore those slots instead of throwing validation errors.

Reply:
It is still happening in `_slot_schema`. Sorry, I didn't describe it properly.
Previously `_slot_schema` method did two things:
1. Add extra slots (i.e. for `DynamicServiceHandler`)
2. Replace each validator with `{"value": validator}`, adding `ALLOW_EXTRA`

This PR breaks down this into two separate methods, allowing us to override the first part only. It would give us the universal way to check which slots the intent expects, and avoid code duplication.